Show list-all checkbox in Apps view

andresmgot opened the pull request:
Issue ref: #481
Continuation of #557 

Now that tiller-proxy support status filtering, add the possibility to show all the apps regardless of the status from the UI:

![list-all](https://user-images.githubusercontent.com/4025665/44860378-52cbb380-ac76-11e8-8f05-0162edaaad25.gif)

andresmgot commented on dashboard/src/actions/apps.test.tsx:
```diff
@@ -0,0 +1,64 @@
+
+describe("fetches applications", () => {
+  beforeEach(() => {
+    App.listApps = jest.fn(() => []);
```
well, yes, I don't see a use case in which the test should use the real `listApp` between tests but it's better to restore its functionality at least at the end.

andresmgot commented on dashboard/src/components/AppList/AppList.tsx:
```diff
@@ -11,44 +11,49 @@ import AppListItem from "./AppListItem";
 
 interface IAppListState {
   filter: string;
+  listAll: boolean;
```
true, at the beginning I was using the state but then I changed the approach and forgot to update this. Good catch!

andresmgot commented:
thanks for the review @prydonius!

prydonius commented on dashboard/src/actions/apps.test.tsx:
```diff
@@ -0,0 +1,67 @@
+  const listApps = App.listApps;
+  let listAppsMock: jest.Mock;
+  beforeEach(() => {
+    App.listApps = jest.fn(() => []);
```
not a big deal, but IMO this is more readable as:

```
listAppsMock = jest.fn(() => []);
App.listApps = listAppsMock
```

prydonius commented on dashboard/src/actions/apps.test.tsx:
```diff
@@ -0,0 +1,67 @@
+});
+
+describe("fetches applications", () => {
+  const listApps = App.listApps;
```
we should call this `listAppsOrig` or something to indicate it's the original.

Angelmmiguel commented:
@arapulido @andresmgot sorry, I think I'm missing a bit of the context. Does the `--all` options refer to display the apps from all namespaces?

andresmgot commented:
> Does the --all options refer to display the apps from all namespaces?

@Angelmmiguel it refers to the status of the helm releases. By default, the `helm list` command line tool omits apps that are not in the status "deployed" or "failed". If you specify the flag `helm list --all` it returns all the applications even if they are in a unusable status (like "deleted").

arapulido commented:
I still think that "Show all apps" is as confusing as "List all" (it basically has the same problem). What it means All? Why am I not seeing them all already?

If you think a tooltip is confusing, I would change it to "Show also releases with statuses other than DEPLOYED or FAILED". @Angelmmiguel ideas?

Angelmmiguel commented:
Okey! I agree it's a bit confusing for users because `List/Show all apps` seems to be out of context: _I have already a list of apps, so what does the "all" collection include?_.

I believe we can turn the tables and display a marked checkbox that says `[x] Show only running apps` . The `running` adjective excludes the other statuses (Pending_*|Deleted_*), so I think it's a good word for this filter. Also, setting it as "active" shows the user that it affects to the current app list.

On the other hand, the current location of the checkbox seems to be related with the `Deploy App` button and not the current list. I'd move it right after the search box.

What do you think @arapulido @andresmgot ?

andresmgot commented:
@arapulido @Angelmmiguel I've moved the checkbox to the left and changed the message:

<img width="1106" alt="screen shot 2018-09-04 at 17 41 22" src="https://user-images.githubusercontent.com/4025665/45042012-057e8600-b06a-11e8-9ac7-99a891a64def.png">

PTAL

arapulido commented:
LGTM Thanks!

Angelmmiguel commented:
I'm fine with both solutions. The only advantage of set it as `checked` is that it has a heavier visual weight in the page, so users will notice it. However, I agree it's more complex than the other option. So, yea, let's use `Show deleted apps` :+1:
